cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java

```diff
@@ -0,0 +1,344 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.ballerina.cli.cmd.Constants.PUSH_COMMAND;
+import static io.ballerina.runtime.util.RuntimeConstants.SYSTEM_PROP_BAL_DEBUG;
+import static org.ballerinalang.tool.LauncherUtils.createLauncherException;
+import static org.wso2.ballerinalang.programfile.ProgramFileConstants.IMPLEMENTATION_VERSION;
+import static org.wso2.ballerinalang.programfile.ProgramFileConstants.SUPPORTED_PLATFORMS;
+import static org.wso2.ballerinalang.util.RepoUtils.getRemoteRepoURL;
+
+/**
+ * This class represents the "ballerina push" command.
+ *
+ * @since 2.0.0
+ */
+@CommandLine.Command(name = PUSH_COMMAND, description = "push modules and binaries available locally to "
+        + "Ballerina Central")
+public class PushCommand implements BLauncherCmd {
+    private static PrintStream outStream = System.err;
+
+    @CommandLine.Parameters
+    private List<String> argList;
+
+    @CommandLine.Option(names = {"--help", "-h"}, hidden = true)
+    private boolean helpFlag;
+
+    @CommandLine.Option(names = "--debug", hidden = true)
+    private String debugPort;
+
+    @CommandLine.Option(names = "--repository", hidden = true)
+    private String repositoryHome;
+
+    @CommandLine.Option(names = {"--skip-source-check"}, description = "skip checking if source has changed")
+    private boolean skipSourceCheck;
+
+    @CommandLine.Option(names = "--experimental", description = "enable experimental language features")
+    private boolean experimentalFlag;
+
+    private Path userDir;
+    private PrintStream errStream;
+    
+    public PushCommand() {
+        userDir = Paths.get(System.getProperty("user.dir"));
+        errStream = System.err;
+    }
+    
+    public PushCommand(Path userDir, PrintStream errStream) {
+        this.userDir = userDir;
+        this.errStream = errStream;
+    }
+
+    @Override
+    public void execute() {
+        if (helpFlag) {
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(PUSH_COMMAND);
+            outStream.println(commandUsageInfo);
+            return;
+        }
+
+        // Get source root path.
+        Path sourceRootPath = userDir;
+
+        // Push command only works inside a project
+        if (!ProjectUtils.isBallerinaProject(sourceRootPath)) {
+            Path findRoot = ProjectUtils.findProjectRoot(sourceRootPath);
+            if (null == findRoot) {
+                CommandUtil.printError(errStream,
+                        "Push command can be only run inside a Ballerina project",
+                        null,
+                        false);
+                return;
+            }
+            sourceRootPath = findRoot;
+        }
+
+        // Enable remote debugging
+        if (null != debugPort) {
```

Comment:
Why do we need this.

Reply:
We set `SYSTEM_PROP_BAL_DEBUG` when `--debug` is in the cmd, and decide ballerina program should be running on debug mode using `SYSTEM_PROP_BAL_DEBUG` property in https://github.com/ballerina-platform/ballerina-lang/blob/master/cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/utils/DebugUtils.java#L22